libnetwork/ipamutils/utils.go

```diff
@@ -21,6 +21,24 @@ var (
 	globalScopeDefaultNetworks = []*NetworkToSplit{{"10.0.0.0/8", 24}}
 )
 
+// IPVersion represents the type of IP address.
+type IPVersion int
+
+// Constants for the IP address types.
+const (
+	IPv4 IPVersion = 4
+	IPv6 IPVersion = 6
+)
```

Comment:
Wondering if we need to look at these; there's a bunch of ipVersion types now (which is already confusing), and it looks like we're not very consistent either;

```bash
git grep -i ' ipversion'
cmd/docker-proxy/proxy.go:// ipVersion refers to IP version - v4 or v6
cmd/docker-proxy/proxy.go:type ipVersion string
cmd/docker-proxy/proxy.go:      ipv4 ipVersion = "4"
cmd/docker-proxy/proxy.go:      ipv6 ipVersion = "6"
libnetwork/drivers/bridge/bridge_test.go:       for _, version := range ipVersions {
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:type ipVersion int
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:  ipvnone ipVersion = iota
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:func getIPVersion(config *networkConfiguration) ipVersion {
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:  return ipVersion
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:  doEnable := func(ipVer ipVersion) error {
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:func getForwardingKernelParam(ipVer ipVersion, iface string) string {
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:func getBridgeNFKernelParam(ipVer ipVersion) string {
libnetwork/drivers/bridge/setup_bridgenetfiltering.go:func isPacketForwardingEnabled(ipVer ipVersion, iface string) (bool, error) {
libnetwork/drivers/bridge/setup_ip_tables.go:           if ipVersion == iptables.IPv4 {
libnetwork/ipam/allocator.go:   var v ipVersion
libnetwork/ipam/utils.go:type ipVersion int
libnetwork/ipam/utils.go:func getAddressVersion(ip net.IP) ipVersion {
libnetwork/iptables/iptables.go:// IPVersion refers to IP version, v4 or v6
libnetwork/iptables/iptables.go:type IPVersion string
libnetwork/iptables/iptables.go:        IPv4 IPVersion = "IPV4"
libnetwork/iptables/iptables.go:        IPv6 IPVersion = "IPV6"
libnetwork/iptables/iptables.go:// IPTable defines struct with IPVersion
libnetwork/iptables/iptables.go:        Version IPVersion
libnetwork/iptables/iptables.go:func GetIptable(version IPVersion) *IPTable {
libnetwork/portmapper/proxy.go:// ipVersion refers to IP version - v4 or v6
libnetwork/portmapper/proxy.go:type ipVersion string
libnetwork/portmapper/proxy.go: ipv4 ipVersion = "4"
libnetwork/portmapper/proxy.go: ipv6 ipVersion = "6"
libnetwork/portmapper/proxy.go: ipVersion ipVersion
libnetwork/portmapper/proxy.go:         return &dummyProxy{addr: addr, ipVersion: version}, nil
libnetwork/portmapper/proxy.go:         return &dummyProxy{addr: addr, ipVersion: version}, nil
libnetwork/portmapper/proxy.go:         return &dummyProxy{addr: addr, ipVersion: version}, nil
```

- some are strings
- difference in _what_ string they are ("4" / "6" / "IPV4" / "IPV6")
- some are ints

FWIW: not sure if we're able to unify them _all_, and what the best location is for a (possibly) exported and canonical one, but my main concern is to introduce a new _exported_ type (once exported it's basically a published API, which complicates moving/changing after that).